### PR TITLE
do not create route in hosted mode

### PR DESCRIPTION
**Pull request by zhiweiyin318**
issue: https://issues.redhat.com/browse/ACM-4652

### CODE REVIEW

File: pkg/klusterlet/clusterinfo/controller.go [hunk 1]
```diff
@@ -84,7 +85,7 @@ func (r *ClusterInfoReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			claimLister:          r.ClaimLister,
 		},
 	}
-	if !r.DisableLoggingInfoSyncer {
+	if !r.DisableLoggingInfoSyncer || r.InstallMode != "Hosted" {
```
Comment (qiujian16) [inline pkg/klusterlet/clusterinfo/controller.go:88]: curious: when do we set DisableLoggingInfoSyncer to true?

Comment (zhiweiyin318) [reply pkg/klusterlet/clusterinfo/controller.go:88]: the controller is imported in  standalone controlplane, and started with DisableLoggingInfoSyncer == true.

Comment (zhiweiyin318) [reply pkg/klusterlet/clusterinfo/controller.go:88]: DisableLoggingInfoSyncer is not a flag of agent binary, let me use it to replace InstallMode.

**Pull request merged**